### ISSUE TITLE
Exception in native call from JS null

### ISSUE THREAD

**Issue by samirakf** [facebook/react-native#23418]
## I have the following exception in my react-native app on Android

<details>
Exception in native call from JS
null
handleException
    ReactRootView.java:564
dispatchDraw
    ReactViewGroup.java:705
draw
    View.java:13715
buildDrawingCache
    View.java:12966
draw
    View.java:13400
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13715
buildDrawingCache
    View.java:12966
draw
    View.java:13400
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13715
buildDrawingCache
    View.java:12966
draw
    View.java:13400
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
draw
    View.java:13715
draw
    FrameLayout.java:467
draw
    HorizontalScrollView.java:1562
draw
    ReactHorizontalScrollView.java:415
draw
    View.java:13596
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
draw
    View.java:13715
draw
    FrameLayout.java:467
draw
    ScrollView.java:1576
draw
    ReactScrollView.java:392
draw
    View.java:13596
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13715
draw
    View.java:13596
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13715
draw
    View.java:13596
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13715
draw
    View.java:13596
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
drawChild
    DrawerLayout.java:1366
dispatchDraw
    ViewGroup.java:2797
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13715
draw
    View.java:13596
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13715
draw
    View.java:13596
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13715
draw
    View.java:13596
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactViewGroup.java:699
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
dispatchDraw
    ReactRootView.java:226
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
draw
    View.java:13715
draw
    FrameLayout.java:467
draw
    View.java:13596
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
draw
    View.java:13594
drawChild
    ViewGroup.java:2928
dispatchDraw
    ViewGroup.java:2797
draw
    View.java:13715
draw
    FrameLayout.java:467
draw
    PhoneWindow.java:2211
drawSoftware
    ViewRootImpl.java:2281
draw
    ViewRootImpl.java:2177
performDraw
    ViewRootImpl.java:2045
performTraversals
    ViewRootImpl.java:1854
doTraversal
    ViewRootImpl.java:989
run
    ViewRootImpl.java:4351
run
    Choreographer.java:749
doCallbacks
    Choreographer.java:562
doFrame
    Choreographer.java:532
run
    Choreographer.java:735
handleCallback
    Handler.java:725
dispatchMessage
    Handler.java:92
loop
    Looper.java:137
main
    ActivityThread.java:5041
invokeNative
    Method.java
invoke
    Method.java:511
run
    ZygoteInit.java:793
main
    ZygoteInit.java:560
main
    NativeStart.java
</details>

## I have the following versions in my React-Native app:

- react-native: 0.57.8
- react: 16.6.3
- Android: 4.4.2

The main problem is that, this exception does not have a certain case, it just happens anywhere in the app. For example, while you've been in the app for a while, and then try to go to a screen, it crashes with this exception.

I have searched a lot about `IllegalViewOperationException`.

https://github.com/facebook/react-native/issues/13984 -> The issue here suggest to remove LayoutAnimation however I've totally removed it from my project and I'm still facing the issue.

[com.facebook.react.uimanager.IllegalViewOperationException: Trying to add unknown view tag ](https://stackoverflow.com/questions/45002568/com-facebook-react-uimanager-illegalviewoperationexception-trying-to-add-unknow)-> the exception here seems to be caused by another reason, in my case I don't have an unknown view tag issue

I have searched a lot about StackOverflowException in react-native, but even in it's GitHub issues, there are zero results.

I've also checked` Exception in native call` error. For example, in this link,[ E/unknown:React: Exception in native call](https://stackoverflow.com/questions/48352667/e-unknownreact-exception-in-native-call), the issue is caused by bundling. However my case is not related to bundling. I can successfully run the app in the emulator and navigate among screens. The exception also occurs in release mode on real devices too.

I'm unable to find an answer including `Exception in native call` ,`IllegalViewOperationException`, and `StackOverflowException` keywords. The exception occurs on Android devices with only 4.x version. It does not occur on 5 and higher Android versions. I guess that it might be related to device's insufficient RAM, or the device might not handle too many renderings in consecutive page navigations. Any ideas on how I can prevent this exception?

**Comment by gaoxiaosong1113** [comment]
TextInput is prompted for this error when it gets focus
![defdeb8b20fe6663e789f41fd45390ba](https://user-images.githubusercontent.com/16183057/53504948-e5e28a80-3aed-11e9-849b-668ebc643148.jpg)

**Comment by murtraja** [comment]
on react-native 57.5 and react-navigation 3.3.0, this (red screen IllegalViewOperation, Exception in Native call from JS) happens consistently when I do the following in Android
  1. create a screen with `createMaterialTopTabNavigator` ( I had 3 tabs, I individually loaded the tabs without TopTabNavigator, it doesn't throw error, this only happens when I use TopTabNavigator)
  2. minimize it (press home button in Android)
  3. resume the application

**Comment by gaoxiaosong1113** [comment]
Here, I use ```react-native-router-flux```. When I open the page with input box, android will prompt this error, but ios does not

**Comment by kelset** [comment]
It seems to me that this issue is a collection of different edge case issues, a few related to libraries and some related to Android old OS versions.

That said, one thing that probably can solve these kind of issue *may* be upgrading to 0.59, in which we use a new version of the JSC - which is probably somewhat related to those native crashes.

If even after upgrading to RN 0.59 you still see these crashes, please open a dedicated issue to a *particular* crash caused by a *specific* reason (with a repro): with the generic descriptions provided above are not helpful for investigating it further.

**Comment by alexkuttig** [comment]
@gaoxiaosong1113 @murtraja did upgrading RN work for you or did you find any other solution? I'm seeing similar crashes in my bugsnag...

**Comment by murtraja** [comment]
@alexkuttig, unfortunately, I wasn't able to find any solution. I haven't upgraded to RN 0.59 either.

**Comment by alexkuttig** [comment]
@murtraja short update - Upgrading RN to 0.59.5 did not solve the problem

**Comment by IndianNative** [comment]
Have any one resolved this issue . I am getting this error in android 8.1  and above whenever I am switching tabs using navigation drawer and all the tabs are maps with different data 
![Screenshot_2019-05-20-02-07-04-398_com where2go](https://user-images.githubusercontent.com/50475442/57987807-75bc2300-7aa4-11e9-8f4b-d7a96a745a47.png)

**Comment by quachsimon** [comment]
I am also running into this issue on 

RN v0.59.9
Android 9.0 (API 28)

Is there any update to this?

![Screenshot_1560982109](https://user-images.githubusercontent.com/20729321/59804791-555ad080-92bd-11e9-8f96-9471d926f118.png)

**Comment by hosseinsalemi** [comment]
any solution?

**Comment by ghostknow** [comment]
How to solve this problem?

**Comment by IndianNative** [comment]
Hi
Can you send me the link. It has not been showing in profile now.
Thanks and Regards
Shashi Saurabh

**Comment by hsnMoghadasi** [comment]
my problem solved with this:
```
cd android
gradlew clean
gradlew assembleDebug --stacktrace
```

i hope that works for you

**Comment by Strate** [comment]
Do someone resolve problem?

**Comment by alexkuttig** [comment]
@Strate it seems to be that there were two different problems, resulting in this screen. 

The first one was, that if you have a very deep stack depth, it could result in that kind of error on older android versions. I managed to get it to work by taking some levels out of my navigation tree. Additionally you could try this one: https://github.com/react-navigation/react-navigation/issues/5843#issuecomment-515014403

The second one was a crash in the android jsc of react native. This was resolved and pushed to react-native 0.60 and 0.59.10. 

Hope that helps :-)

**Comment by lishanxin** [comment]
I am also running into this issue on:

`Alert.alert`

delete it, solved it.
